src/CommunityToolkit.Maui/Views/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,176 @@
+﻿namespace CommunityToolkit.Maui.Views;
+
+using System.Security.Cryptography;
+using System.Text;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Maui.Controls;
+
+/// <summary>Gravatar image source.</summary>
+/// <remarks>Note that <see cref="UriImageSource"/> is sealed and can't be used as a parent!</remarks>
+public class GravatarImageSource : StreamImageSource, IGravatarImageSource
+{
+	const string defaultGravatarImageAddress = "https://www.gravatar.com/avatar/";
+
+	/// <summary>The backing store for the <see cref="CacheValidity" /> bindable property.</summary>
+	public static readonly BindableProperty CacheValidityProperty = BindableProperty.Create(nameof(CacheValidity), typeof(TimeSpan), typeof(UriImageSource), TimeSpan.FromDays(1));
+
+	/// <summary>The backing store for the <see cref="CachingEnabled" /> bindable property.</summary>
+	public static readonly BindableProperty CachingEnabledProperty = BindableProperty.Create(nameof(CachingEnabled), typeof(bool), typeof(UriImageSource), true);
+
+	/// <summary>The backing store for the <see cref="Email" /> bindable property.</summary>
+	public static readonly BindableProperty? EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.DefaultEmail, propertyChanged: OnEmailPropertyChanged);
+
+	/// <summary>The backing store for the <see cref="Image" /> bindable property.</summary>
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.Defaultimage, propertyChanged: OnDefaultImagePropertyChanged);
+
+	int gravatarSize;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+		Stream = new Func<CancellationToken, Task<Stream>>(cancelationToken => DownloadStreamAsync(Uri, cancelationToken));
+		Uri = new Uri(defaultGravatarImageAddress);
```

Comment:
I know I probably created this line but I wasn't sure if this is a good default? Does it behave or result in an exception that will be hidden in `DownloadStreamAsync`? I know we can't make `Uri` nullable because the `IUriImageSource` interface defines it as not nullable.

Reply:
@bijington It does 'appear' to behave, and it's a good "default" if the developer does not provide any email (as it's nullable), and it provides the default image from gravatar.